### PR TITLE
7903837: apidiff: build.sh issues on Windows

### PR DESCRIPTION
**Pull request by jonathan-gibbons**
Please review some `build.sh` and `Makefile` changes to address issues building on Windows.

The primary Windows issue was missing use of `mixed_path` when passing some filenames into `make`.
There is some additional cleanup (for all platforms) to normalize the `DEPS_DIR` and add missing items to the `sanity` and `check_build_vars` targets.

This has been tested on macOS. Can someone with access to a Windows machine test it on Windows, please?

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [CODETOOLS-7903837](https://bugs.openjdk.org/browse/CODETOOLS-7903837): apidiff: build.sh issues on Windows (**Bug** - P3)

### Reviewers
 * [Iris Clark](https://openjdk.org/census#iris) (@irisclark - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/apidiff.git pull/12/head:pull/12` \
`$ git checkout pull/12`

Update a local copy of the PR: \
`$ git checkout pull/12` \
`$ git pull https://git.openjdk.org/apidiff.git pull/12/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12`

View PR using the GUI difftool: \
`$ git pr show -t 12`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/apidiff/pull/12.diff">https://git.openjdk.org/apidiff/pull/12.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/apidiff/pull/12#issuecomment-2374794072)

### CODE REVIEW

**Comment by sormuras** [comment]
> This has been tested on macOS. Can someone with access to a Windows machine test it on Windows, please?

On it. An empty classpath element results in a `::` sequence, which seems to make MSYS2/Cygwin fail the build.

**Review by sormuras — APPROVED** [review]

**Review by sormuras — CHANGES_REQUESTED** [review]
https://github.com/openjdk/apidiff/blob/a10e1426e21978b522d93e1406731240197c2d37/test/junit/JUnitTests.gmk#L33 also needs some extra care to not pass an empty `$(DAISYDIFF_JAR)` element.

**Review by sormuras — APPROVED** [review]

File: test/junit/JUnitTests.gmk [hunk 1]
```diff
@@ -30,7 +30,11 @@ JUnitTest.add-exports =  \
 		--add-exports jdk.jdeps/com.sun.tools.classfile=ALL-UNNAMED
 
 JUnitTest.classpath = \
-		$(BUILDTESTDIR)/JUnitTests/classes:$(APIDIFF_IMAGEDIR)/lib/apidiff.jar:$(JAVADIFFUTILS_JAR):$(DAISYDIFF_JAR):$(HTMLCLEANER_JAR)
+		$(BUILDTESTDIR)/JUnitTests/classes:$(APIDIFF_IMAGEDIR)/lib/apidiff.jar:$(JAVADIFFUTILS_JAR):$(HTMLCLEANER_JAR)
```
Comment (sormuras) [inline test/junit/JUnitTests.gmk:33]:
```suggestion
		$(BUILDTESTDIR)/JUnitTests/classes:$(APIDIFF_IMAGEDIR)/lib/apidiff.jar:$(JAVADIFFUTILS_JAR):$(HTMLCLEANER_JAR):$(EQUINOX_JAR)
```

**Review by sormuras — CHANGES_REQUESTED** [review]

**Review by irisclark — APPROVED** [review]

**Pull request closed without merging**